test/extended/etcd/helpers/helpers.go

```diff
@@ -75,7 +75,9 @@ func CreateNewMasterMachine(ctx context.Context, t TestingT, machineClient machi
 
 func EnsureMasterMachine(ctx context.Context, t TestingT, machineName string, machineClient machinev1beta1client.MachineInterface) error {
 	waitPollInterval := 15 * time.Second
-	waitPollTimeout := 10 * time.Minute
+	// This timeout should be tuned for the platform that takes the longest to provision a node and result
+	// in a Running machine phase.
+	waitPollTimeout := 15 * time.Minute
```

Comment:
I don't think the way this is handled is robust enough.  If you'll want to run this test on metal (it's skipped currently), the boot process alone on some enterprise hardware can take 15 minutes.  I think the metal worst case is probably about 45 minutes.

Do you have any data showing this value will work 100% of the time for the cloud platforms? Even on azure or gcp, I feel like this value isn't the worst case meaning this is still going to fail sometimes.  

What do you think about having this value being quite a bit higher, but failing early in the  Poll condition if machinePhase is Failed? You'll give more time to provision an instance, but still fail early if something unrecoverable happens.

Reply:
@stbenjam That's a good point. 

Although over the past 300 or so runs of this test across all platforms I haven't seen a timeout on node provisioning for AWS or GCP, so based on that observation the 10 min timeout seems to be enough for those two platforms.

All the Azure failures that we've seen so far have their node provisioned around the 10-11 minute mark (from machine creation).
E.g: https://prow.ci.openshift.org/view/gs/origin-ci-test/logs/periodic-ci-openshift-release-master-ci-4.12-e2e-azure-sdn-serial/1551867140221440000

```console
I0726 11:56:00.895641       1 controller.go:179] ci-op-7qjtrydq-45a6f-4mqmr-master-0-clone: reconciling Machine
...
I0726 12:06:10.437503       1 controller.go:426] ci-op-7qjtrydq-45a6f-4mqmr-master-0-clone: going into phase "Running"
```
I haven't seen anything higher which is why I set it for 15 but I'm okay with setting it to 20 mins or higher to provide more cushion.


As for Baremetal, it's already [been enabled](https://github.com/openshift/origin/pull/27294/files#diff-03b29bf92b33772cc9ec978681227c6c4fdfdca04d2279fac287a5ece33f7a4fL365) for these past 300 runs. E.g (https://prow.ci.openshift.org/job-history/origin-ci-test/logs/periodic-ci-openshift-release-master-nightly-4.12-e2e-metal-ipi-serial-ovn-ipv6)

Although I can't attribute those failures to this particular (machine creation)=>(node provisioned) timeout.

The timeout for Baremetal in our test are in the pre-test phase when we try to provision a host ahead of time.
https://github.com/openshift/origin/blob/master/test/extended/etcd/helpers/helpers.go#L351
https://github.com/openshift/origin/blob/master/test/extended/baremetal/helper.go#L117
https://github.com/openshift/origin/blob/master/test/extended/baremetal/helper.go#L174-L193

Other tests that provision extra workers on baremetal also flake on this timeout.
E.g `Managed cluster should grow and decrease when scaling different machineSets simultaneously` is also failing the same as the etcd vertical scaling test here:
https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/openshift_baremetal-operator/226/pull-ci-openshift-baremetal-operator-master-e2e-metal-ipi-serial-ipv4/1551772478584393728

So that's a related but different timeout to fix, and bumping this will have no effect on the baremetal failures that we're currently seeing. That I'll have to track separately.

Reply:
Updated to 25 mins.